### PR TITLE
UI to Move, Merge, and Delete Contacts

### PR DESCRIPTION
**Pull request by kennsippell**
Fix for #162

Backend Changes
* https://github.com/medic/cht-conf/pull/647
* https://github.com/medic/cht-conf/pull/649
* https://github.com/medic/cht-conf/pull/652
* https://github.com/medic/cht-conf/pull/654

Nav Menu:

image

Merge Screen

image

### CODE REVIEW

File: src/lib/queues.ts [hunk 1]
```diff
@@ -35,7 +35,7 @@ export class BullQueue implements IQueue {
   }
 }
 
-export const getMoveContactQueue = () => new BullQueue(
+export const getChtConfQueue = () => new BullQueue(
   WorkerConfig.moveContactQueue,
```
Comment (paulpascal) [inline src/lib/queues.ts:39]: suggestion: should we rename `WorkerConfig.moveContactQueue` to `WorkerConfig.chtConfQueue` to align with the refactored generalization ?

File: src/liquid/place/manage_hierarchy_form.html [hunk 2]
```diff
@@ -67,7 +73,7 @@ <h1 class="subtitle">To</h1>
         pauseOnHover: true,
         type: 'is-success',
         position: 'top-center',
-        message: `<p>Move has been scheduled. Please allow up to 24 hours for processing.</p>`,
+        message: `<p>Operation has been scheduled. Please allow up to 24 hours for processing.</p>`,
```
Comment (paulpascal) [inline src/liquid/place/manage_hierarchy_form.html:76]:
```suggestion
        message: `<p>{{ op | capitalize }} operation has been scheduled. Please allow up to 24 hours for processing.</p>`,
```

File: src/worker/cht-conf-worker.ts [hunk 3]
```diff
@@ -103,19 +105,23 @@ export class MoveContactWorker {
     }
   }
 
+  private static fetchMonitoringApi(instanceUrl: string) {
+    return axios.get(`${instanceUrl}/api/v2/monitoring`);
+  }
+
   private static async moveContact(job: Job): Promise<JobResult> {
```
Comment (paulpascal) [inline src/worker/cht-conf-worker.ts:112]: suggestion: should we rename this function a well to align with the refactored generalization ?

File: src/worker/main.ts [hunk 4]
```diff
@@ -2,15 +2,15 @@ import { config } from 'dotenv';
...
-import { MoveContactWorker } from './move-contact-worker';
+import { ChtConfWorker } from './cht-conf-worker';
 import { WorkerConfig, checkRedisConnection } from '../config/config-worker';
 
 (async () => {
   const { moveContactQueue, redisConnection} = WorkerConfig;
   await checkRedisConnection();
-  MoveContactWorker.processQueue(
+  ChtConfWorker.processQueue(
     moveContactQueue,
```
Comment (paulpascal) [inline src/worker/main.ts:12]: `chtConfQueue` ?

File: src/lib/manage-hierarchy.ts [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+  await RemotePlaceResolver.resolveOne(place, sessionCache, chtApi, { fuzz: true });
+  place.validate();
+
+  const validationError = place.validationErrors && Object.keys(place.validationErrors).find(err => err.startsWith('hierarchy_'));
+  if (validationError) {
+    throw Error(place.validationErrors?.[validationError]);
+  }
+
+  return place.resolvedHierarchy;
+}
```
Comment (paulpascal) [inline src/lib/manage-hierarchy.ts:138]: praise: this is great!

**Review by paulpascal — APPROVED** [review]
Amazing work 🍞 @kennsippell - Thank you for the careful refactoring !

I’ve also tested the changes thoroughly, and everything works as expected. 

Just a few minor suggestions that you can ignore if not necessary for the moment.

File: src/lib/queues.ts (hunk 1 above)
Comment (kennsippell) [reply src/lib/queues.ts:39]: yep! renamed to `chtConfQueueName`

File: src/worker/cht-conf-worker.ts (hunk 3 above)
Comment (kennsippell) [reply src/worker/cht-conf-worker.ts:112]: yep! `processChtConfJob`

File: src/liquid/place/manage_hierarchy_form.html (hunk 2 above)
Comment (kennsippell) [reply src/liquid/place/manage_hierarchy_form.html:76]: Smooth

File: src/worker/main.ts (hunk 4 above)
Comment (kennsippell) [reply src/worker/main.ts:12]: i think this was addressed when i renamed the queue? is that what you're referring to here?

**Pull request merged**